### ISSUE TITLE
Uncaught Error: new BigNumber() number type has more than 15 significant digits

### ISSUE THREAD

**Issue by pointtoken** [MetaMask/metamask-extension#5051]
Metamask  is stuck on a confirmation screen and I can neither cancel or accept:

```
Uncaught Error: new BigNumber() number type has more than 15 significant digits: 1.157920892373162e+77
    at $ (ui.js:1)
    at new P (ui.js:1)
    at n.roundExponential (ui.js:1)
    at ui.js:1
    at ui.js:1
    at ui.js:1
    at ui.js:1
    at ui.js:1
    at Function.mapToProps (ui.js:1)
    at c (ui.js:1)
    at h (ui.js:1)
    at ui.js:1
    at Object.run (ui.js:1)
    at o.onStateChange (ui.js:1)
    at Object.notify (ui.js:1)
    at e.notifyNestedSubs (ui.js:1)
    at o.onStateChange (ui.js:1)
    at Object.notify (ui.js:1)
    at e.notifyNestedSubs (ui.js:1)
    at o.onStateChange (ui.js:1)
    at Object.notify (ui.js:1)
    at e.notifyNestedSubs (ui.js:1)
    at o.onStateChange (ui.js:1)
    at Object.notify (ui.js:1)
    at e.notifyNestedSubs (ui.js:1)
    at o.onStateChange (ui.js:1)
    at Object.notify (ui.js:1)
    at e.notifyNestedSubs (ui.js:1)
    at o.onStateChange (ui.js:1)
    at Object.notify (ui.js:1)
    at e.notifyNestedSubs (ui.js:1)
    at o.notifyNestedSubsOnComponentDidUpdate (ui.js:1)
    at Q (ui.js:1)
    at ui.js:1
    at e.notifyAll (ui.js:1)
    at p.close (ui.js:1)
    at p.closeAll (ui.js:1)
    at perform (ui.js:1)
    at y.perform (ui.js:1)
    at y.perform (ui.js:1)
    at Object.x [as flushBatchedUpdates] (ui.js:1)
    at l.closeAll (ui.js:1)
    at l.perform (ui.js:1)
    at Object.batchedUpdates (ui.js:1)
    at Object.batchedUpdates (ui.js:1)
    at dispatchEvent (ui.js:1)
    at HTMLDocument.o (ui.js:1)
```
 - OS:linux
 - Browser chrome
 - MetaMask Version latest with new ui

**Additional context**
Happened when using https://dai.makerdao.com/ dapp

**Comment by bdresser** [comment]
@pointtoken could you send your [state logs](https://consensys.zendesk.com/hc/en-us/articles/360004133432-How-to-Download-State-Logs-New-UI-) to support@metamask.io and reference this issue number? 

(@danjm @tmashuang might be related to the BN issue with Gitcoin the other day?)

**Comment by danjm** [comment]
fixed with #5060
